src/extensions/default/MDNDocs/main.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2013 - present Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    // Core modules
+    var _                    = brackets.getModule("thirdparty/lodash"),
+        EditorManager        = brackets.getModule("editor/EditorManager"),
+        CSSUtils             = brackets.getModule("language/CSSUtils"),
+        HTMLUtils            = brackets.getModule("language/HTMLUtils"),
+        ExtensionUtils       = brackets.getModule("utils/ExtensionUtils");
+
+    // Extension modules
+    var InlineDocsViewer = require("InlineDocsViewer");
+
+
+    /*
+     * Caches docs promises
+     */
+    var promiseCache = {};
+
+    /**
+     * Lazily loads JSON docs files. Returns a Promise the is resolved with the parsed Object, or
+     * rejected if the file is missing/corrupt.
+     * @param {string} fileName JSON file to load
+     * @return {!$.Promise}
+     */
+    function getDocs(fileName) {
+        if (!promiseCache[fileName]) {
+            var result = new $.Deferred();
+
+            // XXXBramble - load from server vs. fs
+            $.ajax({
```

Comment:
This change seems very specific of mozilla/adobe, should we instead read from a file like before?